### PR TITLE
chore(all)): Replace all Node 14 references with Node 16

### PR DESCRIPTION
**Pull request by biffgaut**
*Issue #, if available:*

*Description of changes:*
This is mainly test structures and documentation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: bf880dba67f3ab3c157a77ebb077a01d641e7ec4
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ee543eca-f40e-4f5e-88ef-21508a40ec77%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: fc52daec615c8673c24b303883e76abd5ca7669b
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=14112e6a-4e24-48c2-aa54-8b5fff63c769%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/use_cases/aws-restaurant-management-demo/test/managerStack.test.ts
Comment (biffgaut) [inline source/use_cases/aws-restaurant-management-demo/test/managerStack.test.ts:1]: This snapshot included the step function ID. A recent PR added a random suffix to the step function ID, so this snapshot was no longer static and this test was invalid.

**Review by georgebearden — APPROVED** [review]
Looks good to me, just one comment/question.

**Pull request merged**